components/rich_text_editor/extensions/emoji/emoji.test.js

```diff
@@ -0,0 +1,94 @@
+import { mount } from '@vue/test-utils';
+import DtRichTextEditor from '../../rich_text_editor.vue';
+import { EditorContent } from '@tiptap/vue-3';
+
+// Wrappers
+let wrapper;
+let editorEl;
+
+// Test Environment
+let propsData;
+let attrs;
+let slots;
+let listeners;
+
+// Constants
+const baseProps = {
+  modelValue: 'init',
+  inputAriaLabel: 'aria-label text',
+  link: true,
+  inputClass: 'qa-editor',
+};
+
+// Helpers
+const _setChildWrappers = () => {
+  editorEl = document.getElementsByClassName('qa-editor')[0];
+};
+
+const _setValue = async (value) => {
+  // await wrapper.setProps({ modelValue: value });
+  editorEl.innerHTML = value;
+  await wrapper.vm.$nextTick();
```

Comment:
Here I still have doubt on the way to do it.

I don't know tiptap but in the documentation it says to uset [set-content](https://tiptap.dev/api/commands/set-content). 

When I tried to use it, It didn't catch even :cat:, thats a suspicious for me to know if we are doing it in the correct way in the first place.

Reply:
So the weird thing is, the actual implementation uses setContent whenever v-model changes and that does trigger inputrules and mark the node. Am not sure why doing this in test environment is not working as expected.